stan/math/prim/mat/fun/opencl_copy.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_OPENCL_COPY_HPP
+#define STAN_MATH_PRIM_MAT_FUN_OPENCL_COPY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/opencl_context.hpp>
+#include <stan/math/gpu/matrix_gpu.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <CL/cl.hpp>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+namespace stan {
+namespace math {
+
+/**
+ * Copies the source Eigen matrix to
+ * the destination matrix that is stored
+ * on the GPU.
+ *
+ * @tparam T type of data in the Eigen matrix
+ * @param src source Eigen matrix
+ * @param dst destination matrix on the GPU
+ *
+ * @throw <code>std::invalid_argument</code> if the
+ * matrices do not have matching dimensions
+ *
+ */
+template <int R, int C>
+void copy(const Eigen::Matrix<double, R, C>& src, matrix_gpu& dst) {
```

Comment:
I'm really sorry I didn't catch this earlier. We want to switch the order of the arguments to be consistent with the rest of the Stan library. The first argument should be what gets filled in (destination in your language) and the second should be what is assigned (the source).

If you found this in another example, please let me know.